### PR TITLE
MISUV-4701 : FE: Request/View refund: Redesign

### PR DESCRIPTION
**Pull request by AshokGitAccount**

### CODE REVIEW

File: app/controllers/CreditAndRefundController.scala [hunk 1]
```diff
@@ -189,11 +189,9 @@ class CreditAndRefundController @Inject()(val authorisedFunctions: FrontendAutho
       case financialDetailsModel: List[FinancialDetailsModel] =>
         val balance: Option[BalanceDetails] = financialDetailsModel.headOption.map(balance => balance.balanceDetails)
         repaymentService.start(user.nino, balance.flatMap(_.getAbsoluteAvailableCreditAmount).getOrElse(BigDecimal(0))).flatMap {
-          case RepaymentJourneyModel(nextUrl) =>
+          case Right(nextUrl) =>
```
Comment (avelx) [inline app/controllers/CreditAndRefundController.scala:192]: pls move logic above into service layer

File: app/services/RepaymentService.scala [hunk 2]
```diff
@@ -17,27 +17,39 @@
...
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 @Singleton
-class RepaymentService @Inject()(val repaymentConnector: RepaymentConnector){
+class RepaymentService @Inject()(val repaymentConnector: RepaymentConnector) {
 
   def start(nino: String, fullAmount: BigDecimal)
-           (implicit headerCarrier: HeaderCarrier): Future[RepaymentJourneyResponseModel] = {
+           (implicit headerCarrier: HeaderCarrier): Future[Either[Throwable, String]] = {
```
Comment (avelx) [inline app/services/RepaymentService.scala:32]: you might pass fullAmount as Option[BigDecimal] here and do extract absolute value, for None value just raise an exception

File: app/services/RepaymentService.scala [hunk 3]
```diff
@@ -17,27 +17,39 @@
...
     Logger("application").debug(s"Repayment journey start with nino: $nino and fullAmount: $fullAmount ")
-    repaymentConnector.start(nino, fullAmount)
+    repaymentConnector.start(nino, fullAmount).map {
+      case RepaymentJourneyModel(nextUrl) =>
+        Right(nextUrl)
+      case RepaymentJourneyErrorResponse(status, message) =>
+        Logger("application").error(s"Repayment journey start error with response code: $status and message: $message}")
```
Comment (avelx) [inline app/services/RepaymentService.scala:38]: pls apply formatting for the log entries, i.e => [CalculationService][getLatestCalculation] ,

File: app/services/RepaymentService.scala [hunk 4]
```diff
@@ -17,27 +17,39 @@
...
+        Left(new InternalError)
+    }.recover { case ex: Exception => Left(ex) }
```
Comment (avelx) [inline app/services/RepaymentService.scala:40]: we need to have log error entry within recover so we can identify this during investigation etc

File: app/services/RepaymentService.scala (hunk 2 above)
Comment (AshokGitAccount) [reply app/services/RepaymentService.scala:32]: done

File: app/controllers/CreditAndRefundController.scala (hunk 1 above)
Comment (AshokGitAccount) [reply app/controllers/CreditAndRefundController.scala:192]: done

File: app/services/RepaymentService.scala (hunk 3 above)
Comment (AshokGitAccount) [reply app/services/RepaymentService.scala:38]: done

File: app/services/RepaymentService.scala (hunk 4 above)
Comment (AshokGitAccount) [reply app/services/RepaymentService.scala:40]: done

**Review by avelx — APPROVED** [review]
approved

**Pull request merged**